sdks/python/apache_beam/transforms/ptransform.py

```diff
@@ -73,27 +73,25 @@ class _PValueishTransform(object):
 
   This visits a PValueish, contstructing a (possibly mutated) copy.
   """
-  def visit(self, node, *args):
-    return getattr(
-        self,
-        'visit_' + node.__class__.__name__,
-        lambda x, *args: x)(node, *args)
-
-  def visit_list(self, node, *args):
-    return [self.visit(x, *args) for x in node]
-
-  def visit_tuple(self, node, *args):
-    return tuple(self.visit(x, *args) for x in node)
-
-  def visit_dict(self, node, *args):
-    return {key: self.visit(value, *args) for (key, value) in node.items()}
+  def visit_nested(self, node, *args):
+    if isinstance(node, (tuple, list)):
+      # namedtuples require unpacked arguments in their constructor,
```

Comment:
It's not clear that this supports subclasses of a namedtuple.  Such subclasses will inherit the _make class method from the namedtuple, but the _make method will produce the namedtuple not the subclass.  Instead, could we test for the existence of _make (as a test of whether we are dealing with a subclass of namedtuple or a direct subclass of tuple, which we assume has the usual tuple constructor) but still invoke the constructor `node.__class__` when dealing with a subclass of a namedtuple? i.e.

```
if isinstance(node, tuple) and hasattr(node.__class__, '_make'):
  # node is an instance of a subclass of a namedtuple.
  return node.__class__(*[self.visit(x, *args) for x in node])
elif isinstance(node, (tuple, list)):
  ...
```

Reply:
Done.